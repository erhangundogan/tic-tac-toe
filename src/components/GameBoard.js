import React from 'react';

const GameBoard = ({ board, onClick }) => {
  const row = (items, ri) => (
    <tr key={ ri }>
      {
        items.map(
          (item, ci) => {
            const id = ri + '' + ci;
            return (<td className="cell" id={ id } key={ id }>{ item }</td>);
          }
        )
      }
    </tr>
  );

  return (
    <div>
      <div className="col-numbers">
        <div>0</div>
        <div>1</div>
        <div>2</div>
      </div>
      <div className="row-numbers-and-board">
        <div className="row-numbers">
          <div>0</div>
          <div>1</div>
          <div>2</div>
        </div>
        <table className="board" onClick={ onClick }>
          <tbody id="board">
            { board.map((r, i) => row(r, i)) }
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default GameBoard;
