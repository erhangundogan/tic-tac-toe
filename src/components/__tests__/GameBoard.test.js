import React from 'react';
import { render } from '@testing-library/react';
import GameBoard from '../GameBoard';

describe('GameBoard', () => {
  it('renders', () => {
    const onClick = jest.fn();
    const board = [
      ['x', 'o', null],
      [null, null, null],
      [null, null, null],
    ];
    const { container } = render(
      <GameBoard board={ board } onClick={ onClick } />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
