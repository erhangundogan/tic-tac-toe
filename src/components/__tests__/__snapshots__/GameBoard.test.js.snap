// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard renders 1`] = `
<div>
  <div
    class="col-numbers"
  >
    <div>
      0
    </div>
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
  <div
    class="row-numbers-and-board"
  >
    <div
      class="row-numbers"
    >
      <div>
        0
      </div>
      <div>
        1
      </div>
      <div>
        2
      </div>
    </div>
    <table
      class="board"
    >
      <tbody
        id="board"
      >
        <tr>
          <td
            class="cell"
            id="00"
          >
            x
          </td>
          <td
            class="cell"
            id="01"
          >
            o
          </td>
          <td
            class="cell"
            id="02"
          />
        </tr>
        <tr>
          <td
            class="cell"
            id="10"
          />
          <td
            class="cell"
            id="11"
          />
          <td
            class="cell"
            id="12"
          />
        </tr>
        <tr>
          <td
            class="cell"
            id="20"
          />
          <td
            class="cell"
            id="21"
          />
          <td
            class="cell"
            id="22"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
